### ISSUE TITLE
PEP complains that CAPS are started too late.

### ISSUE THREAD

**Issue by deadjdona** [esl/MongooseIM#1528]
MongooseIM version: master branch
Installed from: source
Erlang/OTP version: Erlang/OTP 19 [erts-8.3.5] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

Describe the issue: PEP complains that CAPS are not started yet. Can't wait 0.1 sec.
```
2017-10-17 14:13:**55.394** [warning] <0.701.0>@node_pep:complain_if_modcaps_disabled:258 The PEP plugin is enabled in mod_pubsub of host <<"xmpp.******************.com">>. This plugin requires mod_caps to be enabled, but it isn't.
2017-10-17 14:13:**55.526** [debug] <0.300.0>@gen_mod:start_module:125 Module mod_caps started for <<"xmpp.******************.com">>.
```

**Comment by kzemek** [comment]
Can you attach your `ejabberd.cfg` file?

**Comment by deadjdona** [comment]
```
{mod_csi, [{buffer_max, 102400}]},
{mod_time, []},
{mod_caps, []},
...
{mod_pubsub, [
%% {backend, odbc},
    {access_createnode, all},
{access_create, all},
    {host, "pubsub.@HOST@"},
    {item_publisher, false},
{max_subscriptions_node, 1024000},
    {ignore_pep_from_offline, false},
    {last_item_cache, false},
    {max_items_node, 1024},
    {plugins, [<<"flat">>, <<"push">>, <<"pep">>]},
    {pep_mapping, [{"urn:xmpp:microblog:0", "mb"}, {"urn:xmpp:mytest:0", "pep"}]}
]},

```

**Comment by fenek** [comment]
Hi @deadjdona 

Does this problem persists for you? If so, can you please tell me if your are using any per-host configuration? If not, can you please share full module list, so we could try to reproduce the issue locally?

**Comment by deadjdona** [comment]
haven't seen this on latest release, but maybe new instance is faster.

you can conduct fast test - use smallest possible instance with low ram, add pubsub module first in modules list and add mod_caps last in modules list, so pep have some time to complain while caps starting. error either appear or not ^_^

**Comment by fenek** [comment]
Someone will try to reproduce this issue in spare time but since it does not break any functionality and is just an unexpected warning, then the priority is rather low. :)

**Comment by deadjdona** [comment]
looks like i've seen this on latest version too, so it persist.

**Comment by deadjdona** [comment]
latest master:

2017-12-13 18:05:44.123 [warning] <0.1777.0>@node_pep:complain_if_modcaps_disabled:258 The PEP plugin is enabled in mod_pubsub of host <<"xmpp.*************.com">>. This plugin requires mod_caps to be enabled, but it isn't.

**Comment by streaps** [comment]
I have the same problem with MongooseIM 3.0.0 (Ubuntu 18.04 deb package):

`2018-06-25 14:09:30.052 [warning] <0.579.0>@node_pep:complain_if_modcaps_disabled:257 The PEP plugin is enabled in mod_pubsub of host <<"localhost">>. This plugin requires mod_caps to be enabled, but it isn't.
`

**Comment by arcusfelis** [comment]
`gen_mod_deps:add_deps/2` reorders modules randomly.
The way we check if mod_caps is started is not correct.

So, there are two solutions:
- remove the warning.
- make deps/2 callback for mod_pubsub. Make a callback for node_pep. It would start mod_caps automatically.
- add `gen_mod:is_loaded_or_about_to_be_started/2` function or something similar.

**Comment by kzemek** [comment]
Should be fixed in #2029 , can you check that branch?
